services/pdf-ms/lambdas/addPdfToCase.ts

```diff
@@ -1,138 +1,95 @@
+
+const PDF_SSM_PARAMS = params.read(config.pdf.envsKeyName);
 
 const dynamoDbConverter = AWS.DynamoDB.Converter;
 
-export const main = async (event: Record<string, any>) => {
-  const caseData: Case & { user?: Record<string, any>; answersArray: AnswerObject[] } = parseCase(
+export async function main(event: Record<string, any>): Promise<Boolean> {
+  // Convert DynamoDB case data to plain object
+  const unMarshalledCase: Record<string, any> = dynamoDbConverter.unmarshall(
     event.detail.dynamodb.NewImage
   );
-  if (caseData.pdfGenerated) {
-    return;
+
+  if (unMarshalledCase.pdfGenerated === 'yes') {
+    return true;
   }
-  const { formId, PK, SK } = caseData;
+
+  const { PK } = unMarshalledCase;
   const personalNumber = PK.substring(5);
-  const user = await getUser(personalNumber);
-  if (user) {
-    caseData.answers.user = user;
+
+  const [getUserCasesError, userCases] = await to(getUserCases(personalNumber));
+  if (getUserCasesError) {
+    console.error(getUserCasesError);
+    throw getUserCasesError;
   }
 
-  // find changed and newly added values, compared to the last case with the same formId.
-  const relevantCases = await getApplicantCasesByFormId(personalNumber, formId);
-  let changedValues: string[] = [];
-  let newValues: string[] = [];
-  if (relevantCases.length > 1) {
-    const sortedCases = relevantCases
-      .filter(c => c.status.type.includes('submitted'))
-      .filter(c => c.SK !== SK)
-      .sort((c1, c2) => c2.updatedAt - c1.updatedAt);
-    const newestRelevantCase = sortedCases[0];
-    const newAndChangedValues = getNewAndChangedValues(caseData, newestRelevantCase);
-    changedValues = newAndChangedValues.changedValues;
-    newValues = newAndChangedValues.newValues;
-    caseData.answers.valuesChanged =
-      changedValues.length > 0 || newValues.length > 0 ? 'Ja' : 'Nej';
+  const [currentCase, previousCase] = sortCasesByDate(userCases, 2);
+
+  const ssmParams = await PDF_SSM_PARAMS;
+  const currentCaseAnswers = currentCase.forms[ssmParams.recurringFormId].answers;
+  const previousCaseAnswers = previousCase.forms[ssmParams.recurringFormId].answers;
+
+  const { changedValues, newValues } = getNewAndChangedCaseAnswerValues(
+    currentCaseAnswers,
+    previousCaseAnswers
+  );
+
+  const [getUserError, user] = await to(getUser(personalNumber));
+  if (getUserError) {
+    console.error(getUserError);
+    throw getUserError;
   }
 
-  // get the correct template
-  const formType = await getFormType(formId);
-  const templates = JSON.parse((await loadFileFromBucket('templateFiles.json')).toString());
-  const templateFiles = templates[Object.keys(templates).includes(formType) ? formType : 'default'];
+  const pdfJsonValues = {
+    ...arrayToObject(currentCaseAnswers),
+    user,
+    valuesChanged: changedValues.length > 0 || newValues.length > 0 ? 'Ja' : 'Nej',
+  };
+
+  const templates = ssmParams.templatesFilenames;
+  const ekbTemplateFiles = templates['EKB-recurring'];
```

Comment:
hardcoding this to only work for EKB-recurring feels bad, but okay for now, I guess...